libpod/image/utils.go

```diff
@@ -20,7 +20,32 @@ import (
 // a match on name:tag
 func findImageInRepotags(search imageParts, images []*Image) (*storage.Image, error) {
 	_, searchName, searchSuspiciousTagValueForSearch := search.suspiciousRefNameTagValuesForSearch()
-	var results []*storage.Image
+	var (
+		results []*storage.Image
+		foundRW bool
+	)
+
+	// This function prefers read/write images over read/only images, if they match
+	appendImage := func(results []*storage.Image, image *Image) []*storage.Image {
+		if foundRW {
+			// If readwrite image exists ignore readonly images
+			if image.IsReadOnly() {
+				return results
+			}
+			// Append readwrite images exists ignore readonly images
+			return append(results, image.image)
```

Comment:
This, as well as the `return results`, is hard to follow for me.

Just have this function do the `results = append(results)` without the two call sites having to duplicate that.

(Absolutely non-blocking: It would also be nice to have the “business as usual, append” case only once.)